### PR TITLE
[needs review] Added Relationship class to spdx-python tool

### PR DESCRIPTION
**Pull request by Yash-Varshney**
Reference: #138 
# Summary
this P.R. is a part of my communitybridge project (1st evaluation). I am pretty much done with code but it needs to be debugged & reviewed. 
This P.R. adds relationship (parser, builder & writer) to every format (rdf, tag, jsonyamlxml).

When running the tests, the relationship shows ERROR:
(mostly this)
```
doc.add_relationship(relationship.Relationship(relationship=relationship))
AttributeError: 'str' object has no attribute 'Relationship'

``` 

## Files Created/Changed:
1. spdx/relationship.py
2. spdx/parsers/rdf.py
3. spdx/parsers/rdfbuilder.py
4. spdx/parsers/tagvalue.py
5. spdx/parser/tagvaluebuilder.py
6. spdx/parser/jsonyamlxml.py
7. spdx/parser/jsonyamlxmlbuilder.py
8. spdx/writers/rdf.py
9. spdx/writers/tagvalue.py
10. spdx/writers/jsonyamlxml.py
11. tests/test_builder.py

## Status
- [x]  Debugging 

- [ ]  Ready to Merge

## Update
1. Above error (AttributeError) has been resolved from rdf & tagvalue builders (json & others left), also now, all the builders test are passing for relationship
2. parsers for relationships (all formats) are now passing all the tests.

> Used 'Black' formatter for all python files edited.

### CODE REVIEW

File: spdx/parsers/tagvaluebuilders.py [hunk 1]
```diff
@@ -523,9 +522,50 @@ def set_annotation_spdx_id(self, doc, spdx_id):
...
+        """
+        # FIXME: this state does not make sense
+        self.relationship_comment_set = False
+
+    def add_relationship(self, doc, relationship):
+        """
+        Raise SPDXValueError if type is unknown.
+        """
+        self.reset_relationship()
+        doc.add_relationship(relationship.Relationship(relationship=relationship))
```
Comment (ShubhamKJha) [inline spdx/parsers/tagvaluebuilders.py:547]: Here, the `relationship` module object is shadowed by `relationship` local variable.

File: README.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 [5]: https://ci.appveyor.com/api/projects/status/0bf9glha2yg9x8ef/branch/master?svg=true
 [6]: https://ci.appveyor.com/project/spdx/tools-python/branch/master
 
+
```
Comment (rtgdk) [inline README.md:14]: Remove this file diff if no changes here.

Comment (Yash-Varshney) [reply README.md]: thanks. removed it..

File: spdx/parsers/tagvaluebuilders.py (hunk 1 above)
Comment (Yash-Varshney) [reply spdx/parsers/tagvaluebuilders.py:547]: thank you @ShubhamKJha, I'm looking into it. I think I might have to change str 'relationship' everywhere !!

**Comment by Yash-Varshney** [comment]
> Thanks!
> 
> can you use an imperative style for your commit messages be more explicit?
> Right now your commit stream reads a bit like this: https://xkcd.com/1296/
> 
> ```
> 'Added the relationship.py file in spdx'
> 'Added relationship class in document class as well as introduced it …
>     "Added project_details.md in local branch" 
> added
> " updated the parsers for rdf "
> correction in last push
> to be continued ...
>     added relation_subject & related_spdx_element 
>     modified parsers and relationship.py 
> added builders for rdf and tagvalue
>     added license to relationship.py 
> removed relationship_type class in relationship.py file
>     updated daily_work 
> added relationship parser in tagvalue.py
>     added create_entity in rdf 
> added relationship in tag
>     added relationship in jsonyamlxml parser & builder 
> updated xmlparser.py to support relationship
>     added relationship writer for rdf format 
> added relationship writer in tagvalue & jsonyamlxml
>     added tests for relationship in tests/test_buildr.py 
> ready for P.R.
> ```
> 
> So the simplest at this stage is to squash in one clean commit IMHO and use a proper message there.
> Also can your reference a ticket for your work?

@pombredanne thank you for your suggestion, I have squashed all the commits into 1.

**Pull request closed without merging**